nni/experiment/config/experiment_config.py

```diff
@@ -156,3 +164,13 @@ def _validate_canonical(self):
         tuner_cnt = (self.tuner is not None) + (self.advisor is not None)
         if tuner_cnt != 1:
             raise ValueError('ExperimentConfig: tuner and advisor must be set one')
+
+def _load_search_space_file(search_space_path):
+    # FIXME
+    # we need this because PyYAML 6.0 does not support YAML 1.2,
+    # which means it is not fully compatible with JSON
+    content = Path(search_space_path).read_text()
+    try:
+        return json.loads(content)
+    except Exception:
+        return yaml.safe_load(content)
```

Comment:
so the yaml format search space still does not support scientific notation, right?

Reply:
It doesn't.